client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -56,11 +56,14 @@ import org.apache.celeborn.common.util.{JavaUtils, PbSerDeUtils, ThreadUtils, Ut
 import org.apache.celeborn.common.util.FunctionConverter._
 import org.apache.celeborn.common.util.ThreadUtils.awaitResult
 import org.apache.celeborn.common.util.Utils.UNKNOWN_APP_SHUFFLE_ID
+import org.apache.celeborn.common.write.PushFailedBatch
 
 object LifecycleManager {
   // shuffle id -> partition id -> partition locations
   type ShuffleFileGroups =
     ConcurrentHashMap[Int, ConcurrentHashMap[Integer, util.Set[PartitionLocation]]]
+  type ShufflePushFailedBatches =
+    ConcurrentHashMap[Int, ConcurrentHashMap[Integer, util.Set[PushFailedBatch]]]
```

Comment:
Is it possible to use `shuffleId -> Set<PushFailedBatch>` instead of `shuffleId -> (mapId -> Set<PushFailedBatch>`?

Reply:
Update shuffleId to UniqueId,done